lua/nvim_lsp/intelephense.lua

```diff
@@ -11,22 +11,38 @@ local installer = util.npm_installer {
   binaries = {bin_name};
 }
 
+local function is_inside_cwd(path, cwd)
+  local function cb(dir, _)
+    return dir == cwd;
+  end
+
+  local dir, _ = util.path.traverse_parents(path, cb);
+
+  return dir == cwd;
+end
+
 skeleton[server_name] = {
   default_config = util.utf8_config {
     cmd = {bin_name, "--stdio"};
     filetypes = {"php"};
-    root_dir = util.root_pattern("composer.json");
+    root_dir = function (pattern)
+      local cwd  = vim.loop.cwd();
+      local root = util.root_pattern("composer.json", ".git")(pattern);
+
+      -- prefer cwd if it is inside root
+      return is_inside_cwd(root, cwd) and cwd or root;
```

Comment:
I don't really understand this logic. This would spawn a different server for every directory inside of the project. Is that intentional? Do be aware that root_dir doesn't modify the vim working directory, it just informs how to spawn a new server.

Reply:
The intention is to create an instance at the CWD, and use the default `root_pattern` otherwise. If you're viewing a file that is a descendant of CWD, it uses the same instance. If you open a file that is not a descendant of CWD, it uses `root_pattern` to find the project directory.